### PR TITLE
Implement server-side validation of client certs

### PR DESCRIPTION
**Pull request by ivmaykov**
This includes the changes in #173, but I would like to get that merged sooner so our intern gets unblocked.

### CODE REVIEW

**Comment by ryantimwilson** [comment]
FYI, building a real-world nifty server with this PR results in EOF errors when trying to connect to it. Please don't merge this until we figure out these errors.

**Comment by siyengar** [comment]
@ryantimwilson ya we've tracked down the issue with the EOF and will be updating the PR

**Comment by ivmaykov** [comment]
@siyengar I've updated it already. I amended the top commit so you'd need to `git pull -f` if you had an earlier version.

File: nifty-core/src/main/java/com/facebook/nifty/ssl/SslSession.java [hunk 1]
```diff
@@ -65,4 +65,13 @@ public long getEstablishedTime() {
     public X509Certificate getPeerCert() {
         return peerCert;
     }
+
+    public String toString() {
+        return "SslSession(alpn=" + (alpn == null ? "null" : alpn) +
```
Comment (SergeyMakarenko) [inline nifty-core/src/main/java/com/facebook/nifty/ssl/SslSession.java]: There is no need to do check for null, right? I mean you could just write something like  return "SslSession(alpn=" + alpn +...

File: nifty-core/src/main/java/com/facebook/nifty/ssl/SslSession.java [hunk 2]
```diff
@@ -65,4 +65,13 @@ public long getEstablishedTime() {
...
+            ", npn=" + (npn == null ? "null" : npn) +
+            ", version=" + (version == null ? "null" : version) +
+            ", cipher=" + (cipher == null ? "null" : cipher) +
+            ", establishedTime=" + Long.toString(establishedTime) +
+            ", peerCert=" + (peerCert == null ? "null" : peerCert.toString()) + ")";
```
Comment (SergeyMakarenko) [inline nifty-core/src/main/java/com/facebook/nifty/ssl/SslSession.java]: No need to check for null and explicitly calling toString. 
I think the following should do the right thing.
", peerCert=" + peerCert + ")";

**Review by SergeyMakarenko — APPROVED** [review]
lgtm with minor comments

**Pull request merged**